### PR TITLE
renovate: update source import paths on Go module major updates

### PR DESCRIPTION
**Pull request by tklauser**
Try to update source import paths on major updates. In some cases, API
changes might be necessary but due to the source changes, the
appropriate code owners will already be pulled in for reviews on the PR.

### CODE REVIEW

**Review by aanm — CHANGES_REQUESTED** [review]
@tklauser I think we should keep the major updates otherwise we will never be notified about those. We have two options:
a) leave the configuration as is and fix the PRs manually when they show up
b) add `gomodUpdateImportPaths` in renovate see https://docs.renovatebot.com/modules/manager/gomod/#additional-information and fix the PRs manually if renovate can't do it for some reason

**Comment by tklauser** [comment]
> @tklauser I think we should keep the major updates otherwise we will never be notified about those.

That's a good point.

> We have two options:
> a) leave the configuration as is and fix the PRs manually when they show up
> b) add `gomodUpdateImportPaths` in renovate see https://docs.renovatebot.com/modules/manager/gomod/#additional-information and fix the PRs manually if renovate can't do it for some reason

TIL about `gomodUpdateImportPaths`, thanks. I'd probably prefer that option since it will also pull in code owners from the modules using the updated dependency how have probable more context to adjust for possible API updates in the new package version.

**Review by aanm — APPROVED** [review]

File: .github/renovate.json5 [hunk 1]
```diff
@@ -74,6 +74,10 @@
         "go.mod",
         "go.sum"
       ],
+      "postUpdateOptions": [
+	// update source import paths on major updates
+        "gomodUpdateImportPaths",
+      ]
```
Comment (gandro) [inline .github/renovate.json5:80]: @tklauser @aanm Note, there is a missing `,`, c.f. https://github.com/cilium/cilium/issues/23230

Comment (tklauser) [reply .github/renovate.json5:80]: Whoopsie, opened #24109 to fix it. Thanks for the report.

**Pull request merged**